lpc55_isp/src/bin/lpc55_flash.rs

```diff
@@ -236,14 +236,43 @@ fn main() -> Result<()> {
 
     // The target _technically_ has autobaud but it's very flaky
     // and these seem to be the preferred settings
+    //
+    // We initially set the timeout short so we can drain the incoming buffer in
+    // a portable manner below. We'll adjust it up after that.
     let mut port = serialport::new(&cmd.port, cmd.baud_rate)
-        .timeout(Duration::from_millis(1000))
+        .timeout(Duration::from_millis(100))
         .data_bits(DataBits::Eight)
         .flow_control(FlowControl::None)
         .parity(Parity::None)
         .stop_bits(StopBits::One)
         .open()?;
 
+    // Extract any bytes left over in the serial port driver from previous
+    // interaction.
+    loop {
+        let mut throwaway = [0; 16];
+        match port.read(&mut throwaway) {
+            Ok(0) => {
+                // This should only happen on nonblocking reads, which we
+                // haven't asked for, but it does mean the buffer is empty so
+                // treat it as success.
+                break;
+            }
+            Ok(_) => {
+                // We've collected some characters to throw away, keep going.
+            }
+            Err(e) if e.kind() == ErrorKind::TimedOut => {
```

Comment:
nit: there are many ErrorKinds.  Using the fully qualified name helps indicate which one it is.